### PR TITLE
Packaging update

### PR DESCRIPTION
**Pull request by plaplant**
This PR makes some general packaging updates, and drops testing of python3.6 in favor of python3.10.

### CODE REVIEW

File: .github/workflows/publish_to_pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+      - name: Setup Minimamba
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          miniforge-version: latest
+          use-mamba: true
+          python-version: ${{ env.PYTHON }}
+          environment-file: ci/${{ env.ENV_NAME }}.yml
+          activate-environment: ${{ env.ENV_NAME }}
```
Comment (steven-murray) [inline .github/workflows/publish_to_pypi.yaml:29]: In this case, you really don't need to use conda at all, you could just use the `setup-python` action. However, perhaps minimamba is actually faster. I know standard pip is faster than standard conda.

File: setup.py [hunk 2]
```diff
@@ -45,8 +45,7 @@ def package_files(package_dir, subdirectory):
     "package_data": {"hera_opm": data_files},
     "install_requires": ["toml>=0.9.4"],
     "zip_safe": False,
-    "setup_requires": ["pytest-runner", "toml>=0.9.4"],
-    "tests_require": ["pytest", "toml>=0.9.4"],
+    "extras_require": {"test": ["pytest", "pytest-cov"]},
```
Comment (steven-murray) [inline setup.py:48]: is there a reason this isn't in `setup.cfg`?

**Review by steven-murray — COMMENTED** [review]
Looking good, just a few comments/questions

File: .github/workflows/publish_to_pypi.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+name: Publish Python distributions to PyPI and TestPyPI
+
+on: [push, pull_request]
```
[the remark this answers is not in the document]
Comment (plaplant) [reply .github/workflows/publish_to_pypi.yaml:3]: Yeah, that's a good point. I did like the idea of trying to build every push to make sure the package will build, but it's true that in that case we don't have to upload to testpypi. I will remove that below.

File: .github/workflows/publish_to_pypi.yaml (hunk 1 above)
Comment (plaplant) [reply .github/workflows/publish_to_pypi.yaml:29]: Good point. The dependencies are probably lightweight enough that `setup-python` should work and I can use pip to get what I need.

File: .github/workflows/publish_to_pypi.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        run: |
+          conda info -a
+          conda list
+
+      - name: Create environment variable
+        if: startsWith(github.ref, 'refs/tags/') != true
+        run: |
+          wget https://gist.github.com/plaplant/0902f09e59166bac742bbd554f3cd2f9/raw/make_dev_version.sh
+          version=$(bash make_dev_version.sh)
+          echo "SETUPTOOLS_SCM_PRETEND_VERSION=$version" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (plaplant) [reply .github/workflows/publish_to_pypi.yaml:41]: Good point, I'll remove this.

File: setup.py (hunk 2 above)
Comment (plaplant) [reply setup.py:48]: Not a good one, I will move it over.

**Review by steven-murray — APPROVED** [review]
Looks good!

**Pull request merged**